packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -592,6 +592,10 @@ export const DatePickerMixin = (subclass) =>
       if (this.inputElement) {
         this.inputElement.setAttribute('aria-expanded', opened);
       }
+      if (!opened && !this._fullscreen && this.getRootNode().activeElement !== this.focusElement) {
+        // Make sure the field is focused when the dropdown closes
+        this.focus();
```

Comment:
It turned out to be unnecessarily complex to use the overlay's restoreFocusOnClose feature for this:

When you open the dropdown, it's not guaranteed that the field input is focused, in which case a wrong element might get focused on dropdown close. This can happen for example if you open the dropdown using the calendar icon or if you have focus somewhere else on open (added a test for this).

To cover this, we'd need to make sure the field is focused before open to have overlay reference the desired focusable and then have it focused on close.

Instead, we can just explicitly focus the field on close.